metrics/metrics_cli.go

```diff
@@ -0,0 +1,191 @@
+package metrics
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"gopkg.in/yaml.v3"
+)
+
+type Config struct {
+	Host string `yaml:"Host"`
+	Port string `yaml:"Port"`
+
+	Type      string `yaml:"Type"`
+	Count     int    `yaml:"Count"`
+	Frequency int    `yaml:"Frequency"`
+	Random    bool   `yaml:"Random"`
+}
+
+/*
+	defaultType valid values include - "all" "counter" "gauge" "histogram" "summary"
+	defaultCount valid values should be >= 0
+	defaultFreq valid values should be >= 0
+	defaultRand valid values should be boolean
+*/
+var defaultType = "all"
+var defaultCount = 1
+var defaultFreq = 15
+var defaultRand = false
+
+type CommandLine struct{}
+
+func (c *Config) Parse(data []byte) error {
+	return yaml.Unmarshal(data, c)
+}
+
+func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "healthy")
+}
+
+// initConnection handles the metric creation and also updates the metrics via go routines
+// The delegation logic is handled here
+func (cli *CommandLine) initConnection(metricType string, count int, freq int, isRandom bool, conf Config, address string) {
```

Comment:
It doesn't look like `conf` is used here at all.  I would suggest one of three courses of action:

1. Remove the `conf` parameter from this method signature.
2. Remove all other parameters from this method signature and pass all configuration to it via `conf`.
3. Make this method take `conf Config` as its receiver and no parameters.

The latter might be preferable since the `CommandLine` struct doesn't seem to hold any data, it's just got some methods hanging off of it.  Having this method take `Config` as its receiver type means that a caller can set up a `Config` struct with appropriate values and then call this method on it.  It would be functionally equivalent to the second alternative, passing solely the `Config` struct as a parameter since method receivers can be thought of as implicit parameters.